### ISSUE TITLE
Zend\Form\Element\File doesnt work with Annotations and Doctrine

### ISSUE THREAD

**Issue by cbichis** [zendframework/zf2#7096]
Hello,

It seems Zend\Form\Element\File doesn't work if we use along both with Annotations and Doctrine:

```
    /** 
     * @var string
     * @ORM\Column(type="string", length=256, nullable=false)
     * @Annotation\Type("Zend\Form\Element\File")
     * @Annotation\Options({"label":"Image"})
     */
    protected $image;
```
is rendered as:
```
<label><span>Image</span><input name="image" type="text" value=""></label>
```

If we remove the ORM part then form file element is rendered correctly:
```
<label><span>Image</span><input type="file" name="image"></label>
```

Note:

I do have set this methods within entity:
```
    public function getImage() {
        return $this->image;
    }

    public function setImage($image) {
        if (is_array($image) && isset($image['tmp_name'])) {
            $image = $image['tmp_name'];
        }

        $this->image = $image;
        
        return $this;
    }
```

**Comment by mpalourdio** [comment]
have you tried using dev-master of doctrine-orm-module ?

**Comment by cbichis** [comment]
Hi,

It seems to work with dev-master. Thank you!

**Comment by cbichis** [comment]
Actually there is still an issue. Even if now the element is rendered correctly I think is attached automatically a wrong validator as I am getting this error on validation:

    Invalid type given. String expected

Removing the ORM part is making validation works ok.

So I think that fix is only related to rendering part and the validation is still unfixed.

**Comment by cbichis** [comment]
It seems the StringLength validator is added to the InputFilter of the element. This one is triggering the error.

**Comment by mpalourdio** [comment]
The validator checks for a string because of ORM\Column(type="string", length=256, nullable=false).

You can see this behavior in ElementAnnotationsListener#handleValidatorField

**Comment by cbichis** [comment]
Yes, but this make no sense since the element is a File. Definitely this is a bug.

**Comment by mpalourdio** [comment]
What happens if you remove the length constraint in dev-master ?

**Comment by cbichis** [comment]
I found a workaround (until there is a fix in place).

I removed the length from @ORM (your suggestion also). Of course, then the orm:validate-schema started to give me warnings
[Database] FAIL - The database schema is not in sync with the current mapping file.

By changing the field in db to 255 characters (which seems to be the default) the warning is gone now.

I will keep the issue open (as is still an actual bug) but the temp workaround allowed me to continue work.
